### PR TITLE
FELIX-6664 - Replacing String representation with customized Buffered…

### PR DESCRIPTION
**Pull request by DominikSuess**
…Reader (newline behavior slightly changing (newlines inside comments eliminated)

### CODE REVIEW

**Comment by rombert** [comment]
@DominikSuess - I think Carsten should review this. In the meantime, I see that the indentation is a bit off, perhaps you can look into this?

**Comment by DominikSuess** [comment]
> @DominikSuess - I think Carsten should review this. In the meantime, I see that the indentation is a bit off, perhaps you can look into this?

Ah looks like I  accidentially moved the whole block left - fixed now.

File: cm.json/src/test/java/org/apache/felix/cm/json/io/impl/JsonSupportTest.java [hunk 1]
```diff
@@ -190,8 +190,6 @@ public void testSeveralComments() throws IOException {
                 + "  \"a\" : 1,\n"
                 + "  \n"
                 + "  \n"
-                + "\n"
```
Comment (cziegeler) [inline cm.json/src/test/java/org/apache/felix/cm/json/io/impl/JsonSupportTest.java:193]: This seems to be a change in behaviour as line numbers will change after comments are removed

Comment (DominikSuess) [reply cm.json/src/test/java/org/apache/felix/cm/json/io/impl/JsonSupportTest.java]: fixed to retain the linecount

**Review by cziegeler — APPROVED** [review]
lgtm

**Pull request merged**